staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -260,6 +260,14 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope RequestSco
 
 				ae := request.AuditEventFrom(ctx)
 				audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
+			} else {
+				if values := req.URL.Query(); len(values) > 0 {
```

Comment:
this is guarded by `if checkBody`. Looks odd.

Reply:
In the delete handler we call the guarding var `allowsOptions`.

Reply:
Yeah, I'm not sure what the point of these variable is, but it looks like "checkBody" and  "allowOptions" have the same goal here?